### DISCUSSION TITLE
How to register a router with Host matcher?

### DISCUSSION THREAD

**Discussion by yhyzgn** [tokio-rs/axum#2155] [Q&A] [answered]
It seems that matching an url with route's path only, but sometimes we need match the `Host` of a request, such as project `API Gateway Service`, the unified entrance of site with multiple domain and path.

**Comment by davidpdrsn** [comment, marked as answer]
Indeed. Axum only supports routing by path or method. You can write a single catch all handler and check the host name yourself.

**Reply by yhyzgn** [reply]
Okey. 
Thank you.
